public/javascripts/controllers/farmer-ctrl.js

```diff
@@ -154,15 +192,105 @@ angular.module('jasmic.controllers')
                 $scope.newFarm = !$scope.newFarm;
                 $scope.farm = {};
             };
+
+            $scope.editFarmLocation = function(arrayindex) {
+
+                $scope.editFarm = !$scope.editFarm;
+                $scope.farm = $scope.farmer.fr_farms[arrayindex];
+
+                $scope.editingFarm = arrayindex;//set the farm currently being edited
+
+            };
+
             $scope.newCommodityItem = function() {
                 $scope.newCommodity = !$scope.newCommodity;
             };
+
+            $scope.editCommodityItem = function(arrayindex) {
+
+                if (arrayindex == null) document.getElementById("commodities-table").style.visibility='visible'; //if cancel button was clicked
+
+                else document.getElementById("commodities-table").style.visibility='hidden';
```

Comment:
@nick22891 please modify. visibility can be controlled with a boolean and using angular's `ng-if`